File: 118-pascals-triangle/pascals-triangle.cpp
class Solution {//vashu
public:
    vector<vector<int>> generate(int numRows) {
    vector<vector<int>> vashu( numRows);{
        for(int i=0;i<numRows;i++){
            vashu[i]=vector<int>(i+1,1);
            for(int j=1;j<i;j++)
           vashu[i][j] = vashu[i-1][j]+ vashu[i-1][j-1];
        }
    }
   return vashu; }
    
};

File: 118-pascals-triangle/pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> vashu = new ArrayList<>();

        for (int i = 0; i < numRows; i++) {
            List<Integer> row = new ArrayList<>();
            
            for (int j = 0; j <= i; j++) {
                if (j == 0 || j == i) {
                    row.add(1); // first and last element are always 1
                } else {
                    int val = vashu.get(i - 1).get(j - 1) + vashu.get(i - 1).get(j);
                    row.add(val); // sum of two elements above
                }
            }
            
            vashu.add(row); // add row to triangle
        }

        return vashu;
    }
}
